### ISSUE TITLE
Any gap setting other than inner or outer resulting in error

### ISSUE THREAD

**Issue by pearcedom** [Airblader/i3#251]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

Attempting to set increased horizontal gaps using the syntax described in the readme

```
gaps inner <px>
gaps outer <px>
gaps horizontal <px>
```

Inner and outer gaps work fine but the `gaps horizontal` addition results in an error when starting (or re-loading) i3. 

## Expected Behavior

A defined inner and outer gap size *plus* the additional gaps to the left and right.

## Reproduction Instructions

Single window required only, to demonstrate that changing the value of `gaps horizontal` has no effect when re-sourcing config, in addition to the error pop-up when either starting or re-sourcing i3.

## Environment

Output of `i3 --moreversion 2>&-`: 
<pre>
i3 version: Binary i3 version:  4.16-2-g98f7674a (2018-11-04, branch "gaps-next") © 2009 Michael Stapelberg and contributors
(Getting version from running i3, press ctrl-c to abort…)
Running i3 version: 4.16-2-g98f7674a (2018-11-04, branch "gaps-next") (pid 15105)
Loaded i3 config: /home/dominicpearce/.config/i3/config (Last modified: Wed 07 Nov 2018 16:24:10 GMT, 171 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

<pre>
# This file has been auto-generated by i3-config-wizard(1).
# It will not be overwritten, so edit it as you like.
#
# Should you change your keyboard layout some time, delete
# this file and re-run i3-config-wizard(1).
#

# i3 config file (v4)
#
# Please see https://i3wm.org/docs/userguide.html for a complete reference!

set $mod Mod4
exec setxkbmap -layout gb

set $BACKGROUND ~/Pictures/backgrounds/7hmeuED.jpg
exec --no-startup-id feh --bg-fill $BACKGROUND
exec --no-startup-id /usr/bin/wal -i $BACKGROUND
exec --no-startup-id tint2 
exec --no-startup-id compton  

# Font for window titles. Will also be used by the bar unless a different font
# is used in the bar {} block below.
font pango:monospace 8

# This font is widely installed, provides lots of unicode glyphs, right-to-left
# text rendering and scalability on retina/hidpi displays (thanks to pango).
#font pango:DejaVu Sans Mono 8

# Before i3 v4.8, we used to recommend this one as the default:
# font -misc-fixed-medium-r-normal--13-120-75-75-C-70-iso10646-1
# The font above is very space-efficient, that is, it looks good, sharp and
# clear in small sizes. However, its unicode glyph coverage is limited, the old
# X core fonts rendering does not support right-to-left and this being a bitmap
# font, it doesn’t scale on retina/hidpi displays.

# Use Mouse+$mod to drag floating windows to their wanted position
floating_modifier $mod

# start a terminal
bindsym $mod+Return exec gnome-terminal
bindsym $mod+Shift+Return exec i3-sensible-terminal

# kill focused window
bindsym $mod+Shift+q kill

# start dmenu (a program launcher)
bindsym $mod+d exec dmenu_run
bindsym $mod+space	exec rofi -show combi
# There also is the (new) i3-dmenu-desktop which only displays applications
# shipping a .desktop file. It is a wrapper around dmenu, so you need that
# installed.
# bindsym $mod+d exec --no-startup-id i3-dmenu-desktop

# change focus
bindsym $mod+j focus left
bindsym $mod+k focus down
bindsym $mod+l focus up
bindsym $mod+semicolon focus right

# alternatively, you can use the cursor keys:
bindsym $mod+Left focus left
bindsym $mod+Down focus down
bindsym $mod+Up focus up
bindsym $mod+Right focus right

# move focused window
bindsym $mod+Shift+j move left
bindsym $mod+Shift+k move down
bindsym $mod+Shift+l move up
bindsym $mod+Shift+semicolon move right

# alternatively, you can use the cursor keys:
bindsym $mod+Shift+Left move left
bindsym $mod+Shift+Down move down
bindsym $mod+Shift+Up move up
bindsym $mod+Shift+Right move right

# split in horizontal orientation
bindsym $mod+h split h

# split in vertical orientation
bindsym $mod+v split v

# enter fullscreen mode for the focused container
#bindsym $mod+f fullscreen toggle # use F11 instead

# change container layout (stacked, tabbed, toggle split)
bindsym $mod+s layout stacking
bindsym $mod+w layout tabbed
bindsym $mod+e layout toggle split

# toggle tiling / floating 
bindsym $mod+f floating toggle

# change focus between tiling / floating windows | bindsym $mod+space focus mode_toggle

# focus the parent container
bindsym $mod+a focus parent

# focus the child container
#bindsym $mod+d focus child

# switch to workspace
bindsym $mod+1 workspace 1
bindsym $mod+2 workspace 2
bindsym $mod+3 workspace 3
bindsym $mod+4 workspace 4
bindsym $mod+5 workspace 5
bindsym $mod+6 workspace 6
bindsym $mod+7 workspace 7
bindsym $mod+8 workspace 8
bindsym $mod+9 workspace 9
bindsym $mod+0 workspace 10

# move focused container to workspace
bindsym $mod+Shift+1 move container to workspace 1
bindsym $mod+Shift+2 move container to workspace 2
bindsym $mod+Shift+3 move container to workspace 3
bindsym $mod+Shift+4 move container to workspace 4
bindsym $mod+Shift+5 move container to workspace 5
bindsym $mod+Shift+6 move container to workspace 6
bindsym $mod+Shift+7 move container to workspace 7
bindsym $mod+Shift+8 move container to workspace 8
bindsym $mod+Shift+9 move container to workspace 9
bindsym $mod+Shift+0 move container to workspace 10

# reload the configuration file
bindsym $mod+Shift+c reload
# restart i3 inplace (preserves your layout/session, can be used to upgrade i3)
bindsym $mod+Shift+r restart
# exit i3 (logs you out of your X session)
bindsym $mod+Shift+e exec "i3-nagbar -t warning -m 'You pressed the exit shortcut. Do you really want to exit i3? This will end your X session.' -b 'Yes, exit i3' 'i3-msg exit'"

# resize window (you can also use the mouse for that)
mode "resize" {
        # These bindings trigger as soon as you enter the resize mode

        # Pressing left will shrink the window’s width.
        # Pressing right will grow the window’s width.
        # Pressing up will shrink the window’s height.
        # Pressing down will grow the window’s height.
        bindsym j resize shrink width 10 px or 10 ppt
        bindsym k resize grow height 10 px or 10 ppt
        bindsym l resize shrink height 10 px or 10 ppt
        bindsym semicolon resize grow width 10 px or 10 ppt

        # same bindings, but for the arrow keys
        bindsym Left resize shrink width 10 px or 10 ppt
        bindsym Down resize grow height 10 px or 10 ppt
        bindsym Up resize shrink height 10 px or 10 ppt
        bindsym Right resize grow width 10 px or 10 ppt

        # back to normal: Enter or Escape
        bindsym Return mode "default"
        bindsym Escape mode "default"
}

bindsym $mod+r mode "resize"

# Start i3bar to display a workspace bar (plus the system information i3status
# finds out, if available)
#bar {
 #       status_command i3status
#}
#

for_window [class="^.*"] border pixel 0

gaps inner 40
gaps outer 20
gaps horizontal 70
#gaps vertical <px>
#gaps top <px>
#gaps right <px>
#gaps bottom <px>
#gaps left <px>

#new_window 1pixel

#stop mouse centering when window switching
mouse_warping none

#enact a binary split scheme
default_orientation vertical 
for_window [class=".*"] split toggle
</pre>

<pre>
ERROR: CONFIG: Expected one of these tokens: <number>
ERROR: CONFIG: (in file /home/dominicpearce/.config/i3/config)
ERROR: CONFIG: Line 170: gaps inner 40
ERROR: CONFIG: Line 171: gaps outer 20
ERROR: CONFIG: Line 172: gaps outer horizontal 70
ERROR: CONFIG:                      ^^^^^^^^^^^^^
ERROR: CONFIG: Line 173: #gaps vertical 200
ERROR: CONFIG: Line 174: #gaps top <px>
ERROR: FYI: You are using i3 version 4.16-2-g98f7674a (2018-11-04, branch "gaps-next")
ERROR: CONFIG: Expected one of these tokens: 'inner', 'outer'
ERROR: CONFIG: (in file /home/dominicpearce/.config/i3/config)
ERROR: CONFIG: Line 169: gaps inner 40
ERROR: CONFIG: Line 170: gaps outer 20
ERROR: CONFIG: Line 171: gaps horizontal 70
ERROR: CONFIG:                ^^^^^^^^^^^^^
ERROR: CONFIG: Line 172: #gaps vertical <px>
ERROR: CONFIG: Line 173: #gaps top <px>
ERROR: FYI: You are using i3 version 4.16-2-g98f7674a (2018-11-04, branch "gaps-next")
ERROR: CONFIG: Expected one of these tokens: 'inner', 'outer'
ERROR: CONFIG: (in file /home/dominicpearce/.config/i3/config)
ERROR: CONFIG: Line 169: gaps inner 40
ERROR: CONFIG: Line 170: gaps outer 20
ERROR: CONFIG: Line 171: gaps horizontal 70
ERROR: CONFIG:                ^^^^^^^^^^^^^
ERROR: CONFIG: Line 172: #gaps vertical <px>
ERROR: CONFIG: Line 173: #gaps top <px>
ERROR: FYI: You are using i3 version 4.16-2-g98f7674a (2018-11-04, branch "gaps-next")
</pre>

<pre>
- Linux Distribution & Version: Ubuntu 18.04.1 LTS x86_64
- Are you using a compositor (e.g., xcompmgr or compton): compton
</pre>

**Comment by Airblader** [comment]
This has been merged after the 4.16 release so you need a current build from source to get this.

**Comment by basswaver** [comment]
Same behavior in 4.16
```
i3--version
i3 version 4.16 (2018-11-04) © 2009 Michael Stapelberg and contributors
```
```
ERROR: CONFIG: Expected one of these tokens: 'inner', 'outer'
ERROR: CONFIG: (in file /home/zero/.config/i3/config)
ERROR: CONFIG: Line 177: gaps inner 4
ERROR: CONFIG: Line 178: gaps outer 14
ERROR: CONFIG: Line 179: gaps horizontal 10
```
